.github/workflows/versionincrement.yaml

```diff
@@ -0,0 +1,87 @@
+---
+name: Version Increment
+
+on:
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: Log level
+        required: true
+        default: warning
+        type: choice
+        options:
+          - info
+          - warning
+          - debug
+
+jobs:
+  plugin-version-increment-sync:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        entry:
+          - {repo: alerting}
+          - {repo: anomaly-detection}
+          - {repo: asynchronous-search}
+          - {repo: common-utils}
+          - {repo: cross-cluster-replication}
+          - {repo: dashboards-reports, path: reports-scheduler}
+          - {repo: geospatial}
+          - {repo: index-management}
+          - {repo: job-scheduler}
+          - {repo: k-NN}
+          - {repo: ml-commons}
+          - {repo: notifications, path: notifications}
+          - {repo: observability, path: opensearch-observability}
+          - {repo: performance-analyzer}
+          - {repo: performance-analyzer-rca}
+          - {repo: security}
+          - {repo: sql}
+        branch:
+          - 'main'
+          - '2.x'
+          - '2.1'
+          - '2.2'
+    steps:
+      - name: Check out OpenSearch repo
+        uses: actions/checkout@v3
+        with:
+          repository: opensearch-project/OpenSearch
+          ref: ${{ matrix.branch }}
+      - name: Fetch OpenSearch version
+        run: |
+          OPENSEARCH_VERSION=$(cat buildSrc/version.properties | grep opensearch | cut -d= -f2 | grep -oE '[0-9.]+')
+          echo "OPENSEARCH_VERSION_NUMBER=$OPENSEARCH_VERSION" >> $GITHUB_ENV
+          OPENSEARCH_VERSION=$OPENSEARCH_VERSION-SNAPSHOT
```

Comment:
I imagine you tested this works? Strange that this is not parsed as an entire variable "OPENSEARCH_VERSION-SNAPSHOT"

Reply:
Yes @dblock here is the sample action 
https://github.com/prudhvigodithi/opensearch-build/runs/7326648261?check_suite_focus=true
within the step Increment Version in dashboard-repors, you should see `./gradlew updateVersion -DnewVersion=3.0.0-SNAPSHOT`, the `3.0.0-SNAPSHOT` is actually the `OPENSEARCH_VERSION`